posthog/test/test_instance_setting_model.py

```diff
@@ -67,3 +69,22 @@ def test_can_retrieve_multiple_settings(db):
         "MATERIALIZED_COLUMNS_ENABLED": True,
         "ASYNC_MIGRATIONS_AUTO_CONTINUE": True,
     }
+
+
+@pytest.mark.parametrize("cached_value", ("true", '["1:cool","abc:123"]', '"a_string"', "123"))
+def test_get_cached_instance_setting(db, cache, cached_value):
```

Comment:
Can we have a test for when the cache or db call takes a long time? do we time out?

I'm a little defensive about the capture endpoint given how key it is.

Reply:
For what it's worth, the redis cache has https://github.com/jazzband/django-redis#socket-timeout options but looks like this [isn't set by default](https://github.com/jazzband/django-redis/blob/9d572cd460e1acc0458dfa32a5137eb55b844c52/django_redis/pool.py#L57)

Let's say we update an AWS security group on the redis instance incorrectly and we have the connection hang, I would prefer to have this timeout quickly and not get the cached value.

It's a bit pedantic and this PR is already quite large so I'd be happy to take the risk and ignore this for now.

Reply:
I'll resolve this for now.

Reply:
I'm happy to set this to a low value (maybe start with 5 seconds?) and test out a timeout in 1-2 unit tests.

Reply:
I've added error handling to the capture endpoint so any exception would cause us to go back to the default configuration. I assume that code would handle timeouts too :+1:

Reply:
If you feel up to it sure! But do feel free to ignore me.

5 seconds sounds good fwiw.

Reply:
If you keep going down, looks like the exception we eventually get is a `redis-py` `TimeoutError`: https://github.com/redis/redis-py/blob/e39c7babdd80ecdd930bed9d201e8fb9187309ac/redis/connection.py#L215

I'll try mocking this and seeing what happens.

Reply:
😅

Reply:
It was too much mocking :sweat_smile: so I limited myself to raising the more proper `redis.exception.TimeoutError` and ensuring we continue with the fallback (configuration value).

Reply:
Also, I've set the timeout values.